src/goto-programs/get_goto_model.cpp

```diff
@@ -0,0 +1,112 @@
+/*******************************************************************\
+
+Module:
+
+Author: Daniel Kroening, kroening@cs.cmu.edu
+
+\*******************************************************************/
+
+#include "get_goto_model.h"
+
+#include <fstream>
+
+#include <util/message.h>
+#include <util/cmdline.h>
+#include <util/language_file.h>
+#include <util/language.h>
+
+#include <langapi/mode.h>
+
+#include "read_goto_binary.h"
+#include "goto_model.h"
+#include "link_goto_model.h"
+#include "goto_convert_functions.h"
+
+static symbol_tablet get_source(
+  const cmdlinet &cmdline,
+  const std::string &filename,
+  message_handlert &message_handler)
+{
+  messaget message(message_handler);
+
+  languaget *language=get_language_from_filename(filename);
+
+  if(language==nullptr)
+  {
+    source_locationt location;
+    location.set_file(filename);
+    message.error().source_location=location;
+    message.error() << "failed to figure out type of file" << messaget::eom;
+    throw 0;
+  }
+
+  language->get_language_options(cmdline);
+  language->set_message_handler(message_handler);
+
+  #ifdef _MSC_VER
+  std::ifstream infile(widen(filename));
+  #else
+  std::ifstream infile(filename);
+  #endif
+
+  if(!infile)
+  {
+    message.error() << "failed to open input file `"
```

Comment:
Contemplated it, but then we'd be sending out filenames that we know don't exist.